packages/server/src/server.ts

```diff
@@ -1,65 +1,154 @@
-import { Elysia } from 'elysia';
-import { cors } from '@elysiajs/cors';
-import { staticPlugin } from '@elysiajs/static';
-import { html } from '@elysiajs/html';
+import { readdir, stat } from 'node:fs/promises';
 
 import { validateParams } from './validateParams';
-import mockAuth from './mockAuth';
+import { getMockSession, refreshToken } from './mockAuth';
 import { env } from './env/server';
 import ContentService from './content-service';
 
 import renderIndex from './render-index';
 import { fetchDriftsmeldinger, fetchMenu, fetchSearch } from './enonic';
-import mockVarslerHandler from './mockVarsler';
+
+import varslerMock from './varsler-mock.json';
 
 const contentService = new ContentService(fetchMenu);
 
-const port = env.PORT || 8089;
-
-const app = new Elysia()
-  .use(cors())
-  .use(staticPlugin())
-  .use(html())
-  .use(mockAuth)
-  .use(mockVarslerHandler)
-  .get('/api/isReady', () => 'OK')
-  .get('/api/isAlive', () => 'OK')
-  .get('/api/driftsmeldinger', () => fetchDriftsmeldinger())
-  .get('/api/sok', ({ query }) =>
-    fetchSearch(query.ord as string).then((results) => ({
-      hits: results.hits.slice(0, 5),
-      total: results.total,
-    })),
-  )
-  .derive((context) => {
-    const validParams = validateParams(context.query);
-    if (!validParams.success) {
-      console.error(validParams.error);
-      throw new Error(validParams.error.toString());
-    }
-    return {
-      data: validParams.data,
-    };
-  })
-  .get('/data/myPageMenu', ({ data }) =>
-    contentService.getMyPageMenu({ language: data.language }),
-  )
-  .get('/data/headerMenuLinks', ({ data }) =>
-    contentService.getHeaderMenuLinks({
-      language: data.language,
-      context: data.context,
-    }),
-  )
-  .get('/', async ({ data, query, request }) =>
-    renderIndex({
-      contentService,
-      data,
-      url: request.url,
-      query,
+const getFiles = async (dir: string): Promise<string[]> => {
+  const files = await readdir(dir);
+  return Promise.all(
+    files.map(async (name) => {
+      const file = dir + '/' + name;
+      const stats = await stat(file);
+
+      return stats.isDirectory() ? await getFiles(file) : file;
     }),
-  )
-  .listen(port);
+  ).then((all) => all.flat());
+};
+
+const files = (await getFiles('./public')).map((file) =>
+  file.replace('./', '/'),
+);
+
+const validParams = (request: Request) => {
+  const url = new URL(request.url);
+  const query = Object.fromEntries(url.searchParams);
+
+  const validParams = validateParams(query);
+  if (!validParams.success) {
+    console.error(validParams.error);
+    throw new Error(validParams.error.toString());
+  }
+
+  return validParams.data;
+};
+
+const index = async (request: Request) => {
+  const url = new URL(request.url);
+  const query = Object.fromEntries(url.searchParams);
+
+  return renderIndex({
+    contentService,
+    data: validParams(request),
```

Comment:
Litt redd for throwingen som kan skje her, kanskje putte den i en egen variabel før renderIndex slik at man kan catche det?

Reply:
`Bun.serve()` fanger errors for oss, og vi kan også definere vår egen handler: https://bun.sh/docs/api/http#error-handling